e2e/testcases/status_enablement_test.go

```diff
@@ -63,16 +66,47 @@ func TestStatusEnabledAndDisabled(t *testing.T) {
 	nt.MustMergePatch(rootSync, `{"spec": {"override": {"statusMode": "enabled"}}}`)
 	nt.WaitForRepoSyncs()
 
-	_, err = nomostest.Retry(120*time.Second, func() error {
-		rg := resourcegroup.Unstructured(configsync.RootSyncName, configsync.ControllerNamespace, id)
-		err := nt.Validate(configsync.RootSyncName, configsync.ControllerNamespace, rg,
-			nomostest.HasStatus())
-		if err != nil {
-			return err
-		}
-		return nil
-	})
+	err = nomostest.WatchObject(nt, kinds.ResourceGroup(),
+		configsync.RootSyncName, configsync.ControllerNamespace,
+		[]nomostest.Predicate{
+			resourceGroupHasStatus,
+			nomostest.HasLabel(common.InventoryLabel, id),
+		},
+		nomostest.WatchTimeout(120*time.Second))
 	if err != nil {
 		nt.T.Fatal(err)
 	}
 }
+
+func resourceGroupHasNoStatus(obj client.Object) error {
+	if obj == nil {
+		return nomostest.ErrObjectNotFound
+	}
+	rg, ok := obj.(*resourcegroupv1alpha1.ResourceGroup)
+	if !ok {
+		return nil
```

Comment:
Could this return a false positive? I assume this predicate should only be used on the ResourceGroup kind and thus could return an error if it's the wrong type

Reply:
Good catch. This a bug. I’ll fix it.

Reply:
Fixed.